modules/dataquery/jsx/react.app.js

```diff
@@ -233,11 +233,12 @@ class DataQueryApp extends Component {
     // Used to save the current query
 
     let filter = this.saveFilterGroup(this.state.filter);
+    let queryName = name.replace(/ /g,'_');
```

Comment:
This seems to be bandaging over the problem without fixing the root cause. Why can't it contain a space? Is there somewhere that needs to be urlencode/decoded that it isn't happening? If so, this would still break with other characters that have special meaning (+, &, etc)

@nicolasbrossard you had something regarding `+` and saved queries on IBIS, right? Is it related?

Reply:
@driusan, @jesscall, @nicolasbrossard it seems to be used here:
https://github.com/aces/Loris/blob/e7e999dc8f768ae51e630616bd302c5360fc0662/modules/dataquery/ajax/saveQuery.php#L36

and spaces break it.

Reply:
Maybe the frontend fix should be removed and the fix could be done here?

Reply:
@driusan @maltheism 
I've been looking into this and I think the problem runs deeper than the line you linked. 

When saving to the couchDB server with a encoded query name, we start to run into problems with actually retrieving the documents. 

For example I used the query name `Clinical DTT`.
This then gets attributed an id of `jessica.callegaro@mail.mcgill.ca_Clinical+DTT`

However, when attempting to retrieve this document from couchDB, for some reason the `+` is not understood. 

```
curl -X GET http://ccna_dev_user:XXXX@couchhost-dev.acelab.ca:5984/ccna_dev_dqt/"jessica.callegaro@mail.mcgill.ca_Clinical+DTT"

{"error":"not_found","reason":"missing"}
```

Instead of using `+`, its corresponding encoding  needs to be used for retrieval (`%2B`) It looks something like this:

```
curl -X GET http://ccna_dev_user:XXXX@couchhost-dev.acelab.ca:5984/ccna_dev_dqt/"jessica.callegaro@mail.mcgill.ca_Clinical%2BDTT"

{"_id":"jessica.callegaro@mail.mcgill.ca_Clinical+DTT","_rev":"1-d97f5f2aeb78ed68654beb3e299ab134","Meta":{"DocType":"SavedQuery","user":"jessica.callegaro@mail.mcgill.ca","name":"Clinical DTT"},"Fields":{ ... }
```

So once we arrive here https://github.com/aces/Loris/blob/e7e999dc8f768ae51e630616bd302c5360fc0662/modules/dataquery/ajax/GetDoc.php#L33 in the code, it looks something like this:

`$_REQUEST['DocID']` = `jessica.callegaro@mail.mcgill.ca_Clinical DTT`
followed by...
`$docID (encoded)`  = `jessica.callegaro%40mail.mcgill.ca_Clinical+DTT` <---- this will always return undefined!

It's kind of as if we need to double encode the space to actually be able to retrieve a document that contains spaces? idk i spent a lot of time being confused by this, so i hope this explanation made sense? I'd love to hear your opinions though

Reply:
@driusan @maltheism @jesscall I had a similar problem in IBIS: one of our users had saved a query that contained a `+` and the DQT could not retrieve it. I don't think you have to do anything special when saving a query. When you retrieve a query though, you have to call `encodeURIComponent` on the query ID. Look for this ajax call in `modules/dataquery/jsx/react.app.js`:

```javascript
$.ajax(loris.BaseURL + "/AjaxHelper.php?Module=dataquery&script=GetDoc.php&DocID=" + that.state.queryIDs[key][i])
```

It needs to be replaced by:

```javascript
$.ajax(loris.BaseURL + "/AjaxHelper.php?Module=dataquery&script=GetDoc.php&DocID=" + encodeURIComponent(that.state.queryIDs[key][i]))
```

This fixed the issue we had in IBIS.